### DISCUSSION TITLE
Image disappears when using the mouse wheel before any other interaction

### DISCUSSION THREAD

**Discussion by amarandon** [Kitware/trame#481] [Q&A]
Hello, we have a Trame app in which whenever we try to zoom on a visualization using the mouse wheel before doing anything else, the image completely disappears from the screen. However, if we change the orientation of the image by dragging it with the mouse, then we can zoom using the mouse wheel as much as we want and it works fine. This visualization is created with `VtkLocalView`.

In the Chrome debugger, I noticed this warning:
```
[[Violation] Added non-passive event listener to a scroll-blocking 'wheel' event.
Consider marking event handler as 'passive' to make the pag](trame-vtk.js:3518)
```
And also this error:
```
trame-vtk.js:1 
          Can not forward events without a current renderer on the interactor
```
However, I checked the source code of our application and we do seem to assign a renderer to the interactor:
```python
        self.renderer = vtkRenderingCore.vtkRenderer()
        self.window = vtkRenderingCore.vtkRenderWindow()
        self.window.AddRenderer(self.renderer)
        self.interactor = vtkRenderingCore.vtkRenderWindowInteractor()
        self.interactor.SetRenderWindow(self.window)
```
I'm new to Trame and VTK but I'm familiar with Python and web development so I thought I might be able to debug this issue myself. I found a handler for the wheel event was added in `vtk.js`, but I don't know what would be the workflow to modify `vtk.js` and see the result of my modifications in our Trame app. I would need some guidance regarding the JavaScript build pipeline used by Trame. I tried editing the minified JS code inside the install directory of the `trame-vtk` Python package, but obviously that didn't take me very far.

**Comment by jourdain** [comment]
Indeed that is an issue with vtk.js and the wheel handling. I think that is because the wheel has no coordinates, and we can't poke the proper renderer (just a guess).

The best approach will be to reproduce and fix it in vtk.js. 
Then, once that's done, we can update `vue-vtk-js` and `trame-vtk` to use the new version.

HTH

**Reply by marospekarik** [reply]
I am on vtk.js v30.5 and still experiencing this issue.

**Reply by jourdain** [reply]
Indeed until it is tracked/tackled on the vtk.js side, it won't happen. ;-)

**Reply by marospekarik** [reply]
Can you help with any possible workaround until the fix @jourdain?
Thanks

**Reply by jourdain** [reply]
It needs investigation and without [dedicated support](https://www.kitware.com/trame/support/), we can not make it a priority.

**Comment by marospekarik** [comment]
I found a workaround - adding a dummy actor and removing it immediately. This happened only when I loaded the OBJ file with the OBJ Loader. @amarandon , maybe it can serve as a workaround for now.

**Comment by marospekarik** [comment]
Oh, none of this is necessary, make sure you reset the camera of the renderer!
